src/core.js

```diff
@@ -432,17 +430,20 @@ function(i, name) {
 });
 
 function isArraylike( obj ) {
+	var type, length;
+
+	if ( obj == null ||
+		( type = jQuery.type( obj ) ) === "function" ||
+		type === "number" ||
+		jQuery.isWindow( obj ) ) {
+		return false;
```

Comment:
This is very hard to read. Please assign `type` in its declaration, and remove the `type === "number"` check completely. Also, it may be smaller to assign `length` in its declaration as well (e.g., `length = obj != null && "length" in obj && obj.length`).